File: Queue.cpp
#include <iostream>
using namespace std;
#define SIZE 10
class Queue
{
public:
    int *arr;
    int capacity;
    int front;
    int back;

public:
    Queue(int size = SIZE);
    ~Queue();

    void addQueue(int x);
    void deleteQueue();
    int isFull();
    int isEmpty();
    void display();
    Queue *head;
};

Queue::Queue(int size)
{
    arr = new int[size];
    capacity = size;
    front = -1;
    back = -1;
}

Queue::~Queue()
{
    delete[] arr;
}

int Queue::isFull()
{
    return back == capacity - 1 ? EXIT_FAILURE : EXIT_SUCCESS;
}

int Queue::isEmpty()
{
    return back == -1 ? EXIT_FAILURE : EXIT_SUCCESS;
}

void Queue::addQueue(int x)
{
    if (isFull())
    {
        cout << "Queue Overflow" << endl;
        exit(EXIT_FAILURE);
    }
    else
    {
        cout << "Insertion " << endl;
        arr[++back] = x;
    }
}
void Queue::deleteQueue()
{
    if (isEmpty())
    {
        cout << "Queue UnderFlow" << endl;
        exit(EXIT_FAILURE);
    }
    else
    {
        cout << "Deletion" << endl;
        ++front;
    }
}
void Queue::display()
{
    for (int i = front; i <= back; i++)
    {
        cout << arr[i] << endl;
    }
}
int main()
{
    Queue a;
    a.addQueue(1);
    a.addQueue(2);
    a.addQueue(3);
    a.deleteQueue();
    a.addQueue(4);
    a.display();
    return 0;
}

File: queue.py
class queue:
    front = 0
    back = -1

    def __init__(self, size) -> None:
        self.que = [None]*(size+1)
        self.capacity = size

    def enqueue(self,data) -> int:

        if(not self.overflow()):
            queue.back +=1
            self.que[queue.back]=data
            print("Insert ",data)
        else:   
            print("Overflow")

    def dequeue(self)->None:
        if(not self.underflow()):           
            print("Popped") 
            queue.front +=1
        else:
            print("UnderFlow")

    def overflow(self) -> bool:
        return True if(queue.back==self.capacity) else False
    
    def underflow(self) -> bool:
        return True if(queue.back==-1) else False
    
    def display(self)->None:
        for i in range (0,self.capacity):
            print(self.que[i])


obj = queue(3)
obj.enqueue(1)
obj.enqueue(2)
obj.enqueue(3)
obj.enqueue(4)
obj.dequeue()
obj.enqueue(5)
obj.dequeue()
obj.display()